### ISSUE TITLE
meet crash

### ISSUE THREAD

**Issue by lazyjean** [rs/SDWebImage#2025]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | iOS
 Platform Version        | 10.0.2
 SDWebImage Version      | 4.1.0
 Integration Method      | cocoapods
 Xcode Version           | Xcode8
 Repro rate              | only once
 Repro with our demo prj | NO

### Issue Description and Steps
Crashed: NSOperationQueue 0x1702225a0 :: NSOperation 0x170e44950 (QOS: DEFAULT)
0  libobjc.A.dylib                0x18c3db324 cache_getImp + 4
1  libobjc.A.dylib                0x18c3d0c80 lookUpImpOrForward + 284
2  libobjc.A.dylib                0x18c3db258 _objc_msgSend_uncached + 56
3  SDWebImage                     0x1008a2a10 -[SDWebImageDownloaderOperation URLSession:dataTask:didReceiveData:] (SDWebImageDownloaderOperation.m:301)
4  SDWebImage                     0x1008a0868 -[SDWebImageDownloader URLSession:dataTask:didReceiveData:] (SDWebImageDownloader.m:304)
5  CFNetwork                      0x18e141634 __67-[NSURLSession delegate_dataTask:didReceiveData:completionHandler:]_block_invoke.242 + 36
6  Foundation                     0x18e4517e4 __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 16
7  Foundation                     0x18e396358 -[NSBlockOperation main] + 96
8  Foundation                     0x18e386954 -[__NSOperationInternal _start:] + 620
9  Foundation                     0x18e453b90 __NSOQSchedule_f + 228
10 libdispatch.dylib              0x18c8191c0 _dispatch_client_callout + 16
11 libdispatch.dylib              0x18c827444 _dispatch_queue_serial_drain + 928
12 libdispatch.dylib              0x18c81c9a8 _dispatch_queue_invoke + 652
13 libdispatch.dylib              0x18c82938c _dispatch_root_queue_drain + 572
14 libdispatch.dylib              0x18c8290ec _dispatch_worker_thread3 + 124
15 libsystem_pthread.dylib        0x18ca212c8 _pthread_wqthread + 1288
16 libsystem_pthread.dylib        0x18ca20db4 start_wqthread + 4

**Comment by dreampiggy** [comment]
Duplicated issue related to `imageData` multithreads crash : #1998
